torchvision/ops/boxes.py

```diff
@@ -112,8 +113,16 @@ def clip_boxes_to_image(boxes, size):
     boxes_x = boxes[..., 0::2]
     boxes_y = boxes[..., 1::2]
     height, width = size
-    boxes_x = boxes_x.clamp(min=0, max=width)
-    boxes_y = boxes_y.clamp(min=0, max=height)
+
+    if torchvision._is_tracing():
+        boxes_x = torch.max(boxes_x.to(torch.float32), torch.tensor(0., device=boxes.device))
```

Comment:
just wondering, do we need to cast the boxes to `float32` here, or can we just use the `boxes_x` dtype while constructing the scalar tensor?

Reply:
The error is: RuntimeError: result type Float can't be cast to the desired output type Long